### ISSUE TITLE
[a11y] screen reader rect doesn't change position after scroll

### ISSUE THREAD

**Issue by IgorKhramtsov** [flutter/flutter#157718]
### Steps to reproduce

Google Chrome (version 130) on macos with voiceover was used to reproduce.

1. Run provided reproduction code in web
2. Enable screen reader
3. Using screen reader navigation focus text
4. Scroll the page down

### Expected results

Screen reader rect position gets updated to match position of the text

### Actual results

Screen reader rect position stays the same

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: Scaffold(
        appBar: AppBar(
          backgroundColor: Theme.of(context).colorScheme.inversePrimary,
          title: const Text('Flutter Demo Home Page'),
        ),
        body: Center(
          child: SingleChildScrollView(
            child: Semantics(
              explicitChildNodes: true,
              child: Column(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  Semantics(
                    child: Column(
                      children: [
                        const SizedBox(height: 200),
                        Image.network(
                            'https://storage.googleapis.com/cms-storage-bucket/images/Flutter_324_Header_v01.width-635.png'),
                        const Text('some text'),
                        const SizedBox(height: 2000),
                      ],
                    ),
                  ),
                ],
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/b13b41e6-11ac-44b4-843a-1f197ddbf4e9

</details>

### Logs

<details open><summary>Logs</summary>

```console
flutter run -d chrome
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...             14.5s
This app is linked to the debug service: ws://127.0.0.1:61117/DwiSfb9i-xo=/ws
Debug service listening on ws://127.0.0.1:61117/DwiSfb9i-xo=/ws

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

A Dart VM Service on Chrome is available at: http://127.0.0.1:61117/DwiSfb9i-xo=
The Flutter DevTools debugger and profiler on Chrome is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:61117/DwiSfb9i-xo=

```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.24.3, on macOS 14.6.1 23G93 darwin-x64, locale en-GE)
    • Flutter version 3.24.3 on channel stable at /Users/igor_khramtsov/work/tools/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2663184aa7 (7 weeks ago), 2024-09-11 16:27:48 -0500
    • Engine revision 36335019a8
    • Dart version 3.5.3
    • DevTools version 2.37.3

[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/igor_khramtsov/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/to/macos-android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16A242d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915915-b509.11)

[✓] IntelliJ IDEA Community Edition (version 2023.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 77.0.1
    • Dart plugin version 233.13135.65

[✓] VS Code (version 1.93.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.96.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 14.6.1 23G93 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 130.0.6723.70
    ! Error: Browsing on the local area network for Igor’s iPhone. Ensure the device is unlocked and attached with a cable or associated with the same local
      area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by huycozy** [comment]
Thanks for the report. I can also reproduce this issue on Flutter master `3.27.0-1.0.pre.277` on Web app. This looks similar to https://github.com/flutter/flutter/issues/146774 which was fixed before. For macOS desktop app, VoiceOver can't read widgets inside the app.

**Comment by chunhtai** [comment]
I don't think we have control over this, cc @yjbanov is there a way to inform the browser the dom rect has changed?

**Comment by huycozy** [comment]
I assume this issue is being forwarded to Web team, right?

**Comment by chunhtai** [comment]
right, for I think the github UI was bugged, I didn't meant to remove all labels.

**Comment by yjbanov** [comment]
I will double-check, but I've seen this happen even on non-Flutter websites. The screen reader sometimes simply fails to update its focus ring position. Perhaps the problem is that you have the screen reader on, but you are not using the screen reader to actually scroll, but the trackpad/mouse instead. I imagine that when the scrolling action is not taken by the screen reader there needs to be a different channel for the browser to notify the screen reader about the scroll position change.

**Comment by IgorKhramtsov** [comment]
> I will double-check, but I've seen this happen even on non-Flutter websites. The screen reader sometimes simply fails to update its focus ring position. 

A note on that—this indeed happens sometimes on any app, but with the configuration I provided, it happens consistently, and after changing the configuration a bit (removing a semantics widget or a nested column, if I remember correctly), this will not happen anymore.

**Comment by flutter-zl** [comment]
https://github.com/user-attachments/assets/d9911573-481d-47a8-9351-fbbc12eae5bc

I am not seeing the reported issue in dart pad. Screen reader rect position gets updated to match position of the text and image.

Hi @mdebbar, can you help verify?

**Comment by mdebbar** [comment]
@IgorKhramtsov are you still seeing this issue?
